src/actions/getContract.ts

```diff
@@ -587,11 +587,10 @@ type GetWriteFunction<
     : AbiFunction,
   Args = AbiParametersToPrimitiveTypes<TAbiFunction['inputs']>,
   // For making `options` parameter required if `TAccount` or `TChain` is undefined
-  IsOptionsRequired = [IsUndefined<TAccount>, IsUndefined<TChain>] extends
-    | [true, false]
-    | [false, true]
-    ? true
-    : false,
+  IsOptionsRequired = Contains<
+    [IsUndefined<TAccount>, IsUndefined<TChain>],
+    true
+  >,
```

Comment:
Weird that this stopped working in new version of TypeScript. Probably better to use `Or` here instead.

```suggestion
  IsOptionsRequired = Or<
    IsUndefined<TAccount>, IsUndefined<TChain>,
  >,
```

Code for `Or`:

```ts
/**
 * @description Evaluates boolean "or" condition
 * @param T - Type to check
 * @param U - Type to check
 * @example
 * type Result = Or<true, false>
 * //   ^? type Result = true
 */
type Or<T, U> = T extends true ? true : U extends true ? true : false
```

[TypeScript Playground (with tests)](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgUwB5mQYxgFQJ4YDyAZnAL5zFQQhwDkAbsDMgM4x0BQnA9AFR9OcPnAACAEzaYowMDGAQAdnACiDAIYAbAK7qWrOACMIETcnXKARNEtxMS8cwWKhI0WHVR1tHHAC0cPgYcDAQdgAWWADWrmIeXrQAqv6BBMghYZiRmDHCYmjeYGaxMGlwAEps2prwALxwhFAAPDBQ2sgANJRarMgAfLE8PHAjAHoA-CFllazVdSFtyK48nKXBjU04XYl9cPW+aCyK4gat7XCTZ+kAXHDJh8jHp4sXC+e3xD1LnGgY2EHIEhNDZXLpXPp9AAUAEoAHShFQAR10mgBQPBMJ+6CwuDSQJBiy6n00vQhMPhECRKLRxBaiyh0Kxf1xRFpBPaRK+ZLhCORWhpdPaDKZOIF7M6cEUyAYyCg3IpVP5eNpGMZv1FyuBzWJvU5JP6DIVfNRmp1-UxQA)

Reply:
Think `Contains` might be more future-proof (beyond two generics)?

Reply:
Could do an array to `Or` actually!

Reply:
Works! Good points. Fine with either.